marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/endpoints/collector.sjs

```diff
@@ -54,16 +54,16 @@ if(method === 'GET') {
     resp = fn.error(null, "RESTAPI-SRVEXERR", Sequence.from([404, "Not Found", "The requested flow was not found"]));
   } else {
     let stepDoc = flowDoc.steps[step];
-    let baseStep = datahub.flow.step.getStepByNameAndType(stepDoc.name, stepDoc.type);
+    let baseStep = datahub.flow.step.getStepByNameAndType("default-ingest", "INGEST");
```

Comment:
I don't think 'baseStep' is the "default-ingest" step, so we shouldn't hardcode it. In fact, for ingest step, we don't run collector. collector runs only for mapping step

Reply:
@srinathgit 2019-03-30 13:59:45.343 Notice:+in /data-hub/5/endpoints/collector.sjs [javascript]
2019-03-30 13:59:45.808 Info: Status 500: JS-JAVASCRIPT: let combinedOptions = Object.assign({}, baseStep.options, stepDoc.options, flowDoc.options, options); -- Error running JavaScript request: TypeError: Cannot read property 'options' of undefined

https://project.marklogic.com/jira/browse/DHFPROD-2070